server/src/main/java/org/apache/uniffle/server/storage/LocalStorageManager.java

```diff
@@ -94,19 +101,26 @@ public class LocalStorageManager extends SingleStorageManager {
     // We must make sure the order of `storageBasePaths` and `localStorages` is same, or some unit test may be fail
     CountDownLatch countDownLatch = new CountDownLatch(storageBasePaths.size());
     AtomicInteger successCount = new AtomicInteger();
+    ServiceLoader<StorageTypeProvider> loader = ServiceLoader.load(StorageTypeProvider.class);
```

Comment:
Why we need multiple storage type providers? I think one is enough.

Reply:
> I think one is not enough, especially when you need to provide your own provider logic.

Could you help provide some examples about needing multiple providers?

Reply:
I can't agree this, we could use single provider to mark one storage media for one basePath, this is enough.